### PR TITLE
Tickets/dm 47033

### PR DESCRIPTION
**Pull request by eacharles**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,6 +50,7 @@ test: export CM_DATABASE_URL=postgresql://cm-service@localhost:${CM_DATABASE_POR
 test: export CM_DATABASE_PASSWORD=INSECURE-PASSWORD
 test: export CM_DATABASE_SCHEMA=cm_service_test
 test: run-compose
+	cm-service init
```
Comment (fritzm) [inline Makefile:53]: Was this required by something in particular?  The "engine" test fixture already has plumbing in place to reset/init the database (see https://github.com/lsst-dm/cm-service/blob/1b2864745af3dca11786a11873b43a6f7ef08c85/tests/conftest.py#L24)

File: src/lsst/cmservice/handlers/functions.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ async def upsert_spec_block(
     spec_block = spec_block_result.first()
     if spec_block and not allow_update:
         print(f"SpecBlock {key} already defined, skipping it")
```
Comment (fritzm) [inline src/lsst/cmservice/handlers/functions.py:85]: Given updated return value below, are you still "skipping it"?

File: src/lsst/cmservice/handlers/functions.py [hunk 3]
```diff
@@ -167,7 +167,7 @@ async def upsert_script_template(
     script_template = script_template_result.first()
     if script_template and not allow_update:
         print(f"ScriptTemplate {key} already defined, skipping it")
```
Comment (fritzm) [inline src/lsst/cmservice/handlers/functions.py:169]: Same as above, "skipping"?

File: src/lsst/cmservice/handlers/functions.py [hunk 4]
```diff
@@ -215,7 +215,7 @@ async def upsert_specification(
     specification = spec_result.first()
     if specification and not allow_update:
         print(f"Specification {spec_name} already defined, skipping it")
```
Comment (fritzm) [inline src/lsst/cmservice/handlers/functions.py:217]: and one more...

File: tests/routers/test_jobs.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    check_update_methods,
+    create_tree,
+    delete_all_productions,
+)
+
+
+@pytest.mark.asyncio()
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
+async def test_jobs_api(client: AsyncClient) -> None:
+    """Test `/groups` API endpoint."""
```
Comment (fritzm) [inline tests/routers/test_jobs.py:24]: /jobs not /groups?

File: tests/routers/util_functions.py [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+
+    response = await client.get(f"{config.prefix}/{entry_class_name}/get/-1/products")
+    expect_failed_response(response, 404)
+
+
+async def check_queue(
+    client: AsyncClient,
+    entry: models.ElementMixin,
+) -> None:
+    # make and test queue object
```
Comment (fritzm) [inline tests/routers/util_functions.py:664]: maybe add "TODO:" to make this easier to search up

**Review by fritzm — APPROVED** [review]
Looks pretty good -- minor comments left, thanks!

File: Makefile (hunk 1 above)
Comment (eacharles) [reply Makefile:53]: The test were failing with errors that suggested the database wasn't up and this fixed it.  I'll try removing it.

File: src/lsst/cmservice/handlers/functions.py (hunk 2 above)
Comment (eacharles) [reply src/lsst/cmservice/handlers/functions.py:85]: We are not updating it, which seems like "skipping" it.   I could say "leaving it unchanged".

**Pull request merged**